[Feature] 회원 정보 수정 로직 구현

kg0726 opened the pull request:
### 작업 개요
- 사용자가 입력한 내용을 DB에 저장(수정)
- 비밀번호 재인증 로직 구현

### 변경 사항
- User 모델 스키마 작성 및 form 정의  
- 화면에 선택 박스가 출력되도록 모델 스키마 수정
- User 모델 각 필드에서 출력될 이름 변경
- 사용자가 회원 정보 수정 페이지, 회원 탈퇴를 요청하면 비밀번호 재인증이 이루어지도록 views.py의 update, delete 함수 수정
- 비밀번호 재인증 로직 추가
- 비밀번호 인증 템플릿 추가

### 확인 요청
- [x] 로그인 후 회원 정보 페이지로 이동
- [x] 회원 정보 페이지로 이동할 때 비밀번호를 요청하는지 확인
- [x] 비밀번호를 틀렸을 경우 비밀번호가 잘못되었다는 에러 메시지와 함께 다시 비밀번호를 입력하도록 하는지 확인
- [x] 회원 정보 수정 페이지에서 회원 탈퇴 버튼을 클릭했을 때 다시 비밀번호 인증을 요청하는지 확인
- [x] 회원 정보 수정 후 저장 시 DB에 반영되는지 확인
- [x] 회원 정보 수정 후 저장 -> 다시 회원 정보 수정 페이지에 접속하면 수정한 정보가 반영되어 출력되는지 확인

### 참고 사항
- 마이그레이션 반드시 진행 후 리뷰 하셔야 합니다!!
- 비밀번호 재인증 로직 구현 과정에서 새로운 개념들이 많이 등장하여 코드가 무슨 뜻인지 잘 파악할 수 있도록 하기 위해 세세한 주석을 남겨놓았습니다. 그럼에도 이해가 어려울 수 있어 참고 디스커션을 정리하였습니다.
- https://github.com/13aek/finbot/discussions/95

kg0726 committed:
feat: 회원 정보 모델 스키마 작성 및 form 정의

kg0726 committed:
refactor: 화면에 선택 박스가 추가되도록 모델 스키마 수정

kg0726 committed:
refactor User 모델 각 필드에서 출력될 이름 변경

kg0726 committed:
refactor: isort 및 black 적용

kg0726 committed:
feat: 비밀번호 재인증 로직 추가
- 사용자가 회원정보 페이지를 요청하면 비밀번호 인증을 받는 로직 추가
- 사용자가 회원탈퇴를 요청하면 비밀번호 인증을 받는 로직 추가
- 비밀번호 인증 템플릿 추가

HeHe985 reviewed (commented):
종민님 develop 브랜치의 최신 코드에 수정하신것 같지 않습니다 ㅜㅜ
확인 부탁드립니다
base.html, setting.py등 제가 올려서 머지됐던 부분이 없습니다

kg0726 committed:
fix: 비밀번호 인증 로직 수정
- 비밀번호 인증 로직이 제대로 작동하지 않던 현상을 수정하였습니다.
- 이제 사용자가 회원 탈퇴를 요청하고 비밀번호를 인증했을 때, 정상적으로 탈퇴가 이루어집니다.

kg0726 committed:
fix: 최신 develop 반영
- merge시 충돌 방지 및 기존 파일 손상 방지

kg0726 committed:
refactor: isort 및 black 적용

kg0726 commented:
> 종민님 develop 브랜치의 최신 코드에 수정하신것 같지 않습니다 ㅜㅜ 확인 부탁드립니다 base.html, setting.py등 제가 올려서 머지됐던 부분이 없습니다

반영해서 완료했습니다.

HeHe985 reviewed (approved):
모두 확인 완료!
base.html 메서드 -> GET
추후 세션 만료 아닌 세션 삭제 방법 논의할 예정
